src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -421,7 +503,8 @@ internal static void CheckParameterModifiers(
             BindingDiagnosticBag diagnostics,
             bool parsingFunctionPointerParams,
             bool parsingLambdaParams,
-            bool parsingAnonymousMethodParams)
+            bool parsingAnonymousMethodParams,
+            bool extensionReceiverParameter)
```

Comment:
>extensionReceiverParameter [](http://example.com/codeflow?start=17&length=26)

It looks like the last four booleans are all about parameter location and at most one of them can be true at a given time. Consider merging them into a single parameter, probably an enum.